server/tserver/src/test/java/org/apache/accumulo/tserver/log/RecoveryLogsIteratorTest.java

```diff
@@ -0,0 +1,267 @@
+
+  @Before
+  public void setUp() throws Exception {
+    context = createMock(ServerContext.class);
+    logSorter = new LogSorter(context, DefaultConfiguration.getInstance());
+
+    workDir = tempFolder.newFolder();
+    String path = workDir.getAbsolutePath();
+    assertTrue(workDir.delete());
+    fs = VolumeManagerImpl.getLocalForTesting(path);
+    expect(context.getVolumeManager()).andReturn(fs).anyTimes();
+    expect(context.getCryptoService()).andReturn(CryptoServiceFactory.newDefaultInstance())
+        .anyTimes();
+    expect(context.getConfiguration()).andReturn(DefaultConfiguration.getInstance()).anyTimes();
+    replay(context);
+  }
+
+  @After
+  public void tearDown() throws Exception {
+    fs.close();
+  }
+
+  static class KeyValue implements Comparable<KeyValue> {
+    public final LogFileKey key;
+    public final LogFileValue value;
+
+    KeyValue() {
+      key = new LogFileKey();
+      value = new LogFileValue();
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hashCode(key) + Objects.hashCode(value);
+    }
+
+    @Override
+    public boolean equals(Object obj) {
+      return this == obj || (obj instanceof KeyValue && 0 == compareTo((KeyValue) obj));
+    }
+
+    @Override
+    public int compareTo(KeyValue o) {
+      return key.compareTo(o.key);
+    }
+  }
+
+  @Test
+  public void testSimpleRLI() throws IOException {
+    KeyValue keyValue = new KeyValue();
+    keyValue.key.event = DEFINE_TABLET;
+    keyValue.key.seq = 0;
+    keyValue.key.tabletId = 1;
+    keyValue.key.tablet = extent;
+
+    KeyValue[] keyValues = {keyValue};
+
+    Map<String,KeyValue[]> logs = new TreeMap<>();
+    logs.put("keyValues", keyValues);
+
+    ArrayList<Path> dirs = new ArrayList<>();
+
+    createRecoveryDir(logs, dirs, true);
+
+    try (RecoveryLogsIterator rli = new RecoveryLogsIterator(context, dirs, null, null, false)) {
+      while (rli.hasNext()) {
+        Entry<LogFileKey,LogFileValue> entry = rli.next();
+        assertEquals("TabletId does not match", 1, entry.getKey().tabletId);
+        assertEquals("Event does not match", DEFINE_TABLET, entry.getKey().event);
+      }
+    }
+  }
+
+  @Test
+  public void testFinishMarker() throws IOException {
+    KeyValue keyValue = new KeyValue();
+    keyValue.key.event = DEFINE_TABLET;
+    keyValue.key.seq = 0;
+    keyValue.key.tabletId = 1;
+    keyValue.key.tablet = extent;
+
+    KeyValue[] keyValues = {keyValue};
+
+    Map<String,KeyValue[]> logs = new TreeMap<>();
+    logs.put("keyValues", keyValues);
+
+    ArrayList<Path> dirs = new ArrayList<>();
+
+    createRecoveryDir(logs, dirs, false);
+
+    try (RecoveryLogsIterator rli = new RecoveryLogsIterator(context, dirs, null, null, false)) {
+      while (rli.hasNext()) {
+        fail("Finish marker should not be found. Exception should have been thrown.");
+      }
+    } catch (IOException e) {
+      // Expected exception
+    }
+  }
```

Comment:
It may be possible to use assertThrows here, something like:
```suggestion
    assertThrows("Finish marker should not be found", IOException.class, () -> new RecoveryLogsIterator(context, dirs, null, null, false));
  }
```
This code would be slightly different if the expected exception is thrown in the `hasNext()` method instead of in the constructor, but the idea is the same.

Reply:
Made this change in me latest commit.